### PR TITLE
Fix syntax highlighting for `not`

### LINKED ISSUE #12813: `not <anything>` panics in REPL
### Describe the bug

Typing `not` followed by any expression in the REPL panics.

### How to reproduce

Type `not $`, `not (`, `not false`, etc. into REPL.

### Expected behavior

Not panic.

### Screenshots

_No response_

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.93.1                                          |
| major              | 0                                               |
| minor              | 93                                              |
| patch              | 1                                               |
| branch             | main                                            |
| commit_hash        | 3b26c08dab31b98658e0836dd113d1f7af6c8d29        |
| build_os           | linux-x86_64                                    |
| build_target       | x86_64-unknown-linux-gnu                        |
| rust_version       | rustc 1.77.2 (25ef9e3d8 2024-04-09)             |
| rust_channel       | 1.77.2-x86_64-unknown-linux-gnu                 |
| cargo_version      | cargo 1.77.2 (e52e36006 2024-03-26)             |
| build_time         | 2024-05-08 18:43:01 +03:00                      |
| build_rust_channel | release                                         |
| allocator          | mimalloc                                        |
| features           | default, sqlite, system-clipboard, trash, which |
| installed_plugins  | formats, gstat, highlight, inc, polars, query   |

### Additional context

```
not Error:   × Main thread panicked.
  ├─▶ at crates/nu-cli/src/syntax_highlight.rs:87:17
  ╰─▶ byte index 7 is out of bounds of `not $`
         0: 0x62b4c9294094 - rust_begin_unwind
         1: 0x62b4c847c055 - core::panicking::panic_fmt::ha6effc2775a0749c
         2: 0x62b4c8569923 - core::str::slice_error_fail_rt::hae2b0a3e014351e1
         3: 0x62b4c847c867 - core::str::slice_error_fail::he2ff12236fb0c056
         4: 0x62b4c868b180 - <nu_cli::syntax_highlight::NuHighlighter as reedline::highlighter::Highlighter>::highlight::h8d2941798f7e17f9
         5: 0x62b4c915d369 - reedline::engine::Reedline::repaint::h9e33c686510d3945
         6: 0x62b4c9159e0b - reedline::engine::Reedline::read_line::hbd5bb4700b7ef561
         7: 0x62b4c86a82d2 - nu_cli::repl::loop_iteration::hff531597d7240be1
         8: 0x62b4c86a1bae - std::panicking::try::hcdb9f8b95caeefd7
         9: 0x62b4c86ba944 - nu_cli::repl::evaluate_repl::had8f25f31091a173
        10: 0x62b4c8670ec2 - nu::run::run_repl::h3e269baf11b8287c
        11: 0x62b4c865dd95 - nu::main::he0f4b8995bab6346
  help: set the `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

### LINKED ISSUE #12814: problem with ` crates/nu-cli/src/syntax_highlight.rs:87:17` on the latest `main`
### Describe the bug

something is broken on the latest `main` (at least on my mac inside of wezterm)

### How to reproduce

1. paste `not (true and false)` into command line so it will be recorded into history
2. start entering `not (` - bug should appear

You can find my attempts in the video that is attached

https://github.com/nushell/nushell/assets/4896754/70fbc53e-07bd-4f7f-951e-5d200e4a4b6d

### Expected behavior

to behave without bugs

### Screenshots

_No response_

### Configuration

| key                | value                                                      |
| ------------------ | ---------------------------------------------------------- |
| version            | 0.93.1                                                     |
| major              | 0                                                          |
| minor              | 93                                                         |
| patch              | 1                                                          |
| branch             | main                                                       |
| commit_hash        | c54d223ea0b14f180b1793fa539419854c2e014f                   |
| build_os           | macos-aarch64                                              |
| build_target       | aarch64-apple-darwin                                       |
| rust_version       | rustc 1.77.2 (25ef9e3d8 2024-04-09)                        |
| rust_channel       | 1.77.2-aarch64-apple-darwin                                |
| cargo_version      | cargo 1.77.2 (e52e36006 2024-03-26)                        |
| build_time         | 2024-05-07 05:50:32 +00:00                                 |
| build_rust_channel | release                                                    |
| allocator          | mimalloc                                                   |
| features           | dataframe, default, sqlite, system-clipboard, trash, which |
| installed_plugins  | explore, image, inc, polars                                |

### Additional context

Here is my terminal with backtrace:

```
░▒▓ ~   06:27  📬 6
> not (trtrue and false)
true

░▒▓ ~   06:28  📬 6
> not (trtrue and false)
true

░▒▓ ~   06:28  📬 6
> not Error:   × Main thread panicked.
  ├─▶ at crates/nu-cli/src/syntax_highlight.rs:87:17
  ╰─▶ byte index 7 is out of bounds of `not (`
  help: set the `RUST_BACKTRACE=1` environment variable to
        display a backtrace.


░▒▓ ~   06:28  📬 6
> RUST_BACKTRACE=1 nu

░▒▓ ~   06:28  823ms 📬 6
> not (trtrue and false)
true

░▒▓ ~   06:28  📬 6
> not Error:   × Main thread panicked.
  ├─▶ at crates/nu-cli/src/syntax_highlight.rs:87:17
  ╰─▶ byte index 7 is out of bounds of `not (`
         0: 0x10207b718 - _rust_begin_unwind
         1: 0x1022e2fc4 -
      core::panicking::panic_fmt::h1cb43b60f5788132
         2: 0x10013c1ec -
      core::str::slice_error_fail_rt::h4e2622ead69dc193
         3: 0x1022e3560 -
      core::str::slice_error_fail::hc7d32fbcbb1b6342
         4: 0x1002949b8 -
      <nu_cli::syntax_highlight::NuHighlighter as
      reedline::highlighter::Highlighter>::highlight::h9da57f
      61381eae7b
         5: 0x101e6bc28 -
      reedline::engine::Reedline::repaint::h34d9e1cde26a017e
         6: 0x101e68f88 -
      reedline::engine::Reedline::read_line::h33df94a1550d752
      d
         7: 0x1002b2988 -
      nu_cli::repl::loop_iteration::h4c2b2a6c0a77b323
         8: 0x1002abf10 -
      std::panicking::try::ha740e39f612c6486
         9: 0x10026a3ec -
      nu_cli::repl::evaluate_repl::h6e05ce696f2fb0bb
        10: 0x10023e91c -
      nu::run::run_repl::hf11688a360fe35d2
        11: 0x100223fe8 - nu::main::h5f3bb83be07b5395
  help: set the `RUST_BACKTRACE=1` environment variable to
        display a backtrace.

```

### PR DESCRIPTION
**Pull request by IanManske**
# Description
Fixes #12813 where a panic occurs when syntax highlighting `not`. Also fixes #12814 where syntax highlighting for `not` no longer works.

# User-Facing Changes
Bug fix.

### CODE REVIEW

**Comment by WindSoilder** [comment]
Sorry when looking into diff, I don't know how it fixes the issue, it seems just like a rename in a enum variant.
And can you add a test that @kubouch mentioned: https://github.com/nushell/nushell/issues/12813#issuecomment-2102144662

**Comment by IanManske** [comment]
The issue was I used the wrong variable due to shadowing. I just renamed it so that both variables are available.

File: crates/nu-cli/tests/commands/nu_highlight.rs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+use nu_test_support::nu;
+
+#[test]
+fn nu_highlight_not_expr() {
+    let actual = nu!("'not false' | nu-highlight | ansi strip");
```
Comment (sholderbach) [inline crates/nu-cli/tests/commands/nu_highlight.rs:5]: Ok this is clever 😄

Comment (IanManske) [reply crates/nu-cli/tests/commands/nu_highlight.rs:5]: credit goes to @kubouch :+1:

**Comment by sholderbach** [comment]
Does this also close #12813 ?

**Review by WindSoilder — APPROVED** [review]
Thanks!  It looks good to me

**Pull request merged**